tests/suites/helpers.function

```diff
@@ -159,6 +319,17 @@ static void platform_teardown()
 #endif /* MBEDTLS_PLATFORM_C */
 }
 
+#if defined(MBEDTLS_CHECK_PARAMS)
+void mbedtls_param_failed( char* failure_condition, char* file, int line )
+{
+    (void)failure_condition;
+    (void)file;
+    (void)line;
+
+    longjmp( param_fail_jmp, 1 );
```

Comment:
This `longjmp` raises a series of warnings on my compiler. Apparently, the compiler (for -O2) decides to allocate certain locals in registers, which doesn't work with set/longjmp well. Setting the flag from -O2 to -O0 removes the warnings. We don't use the clobbered variables in our callback, but, nevertheless, we should probably avoid warnings in the default build, and there are quite a lot of them.